flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/StopWithSavepoint.java

```diff
@@ -101,6 +101,8 @@ private void handleSavepointCompletion(
             completeOperationAndGoToFinished(savepoint);
         } else {
             if (throwable != null) {
+                // creating the savepoint has failed but job is still running
+                Preconditions.checkState(getExecutionGraph().getState() == JobStatus.RUNNING);
```

Comment:
Why did you introduce this `checkState`? Wouldn't this be caught by the `Executing` state?